File: csharpmabye/TradersWrapper/Models/Contract.cs
// {
//     "id": "string",
//   "factionSymbol": "string",
//   "type": "PROCUREMENT",
//   "terms": {
//         "deadline": "2019-08-24T14:15:22Z",
//     "payment": {
//             "onAccepted": 0,
//       "onFulfilled": 0
//     },
//     "deliver": [
//       { }
//     ]
//   },
//   "accepted": false,
//   "fulfilled": false,
//   "expiration": "2019-08-24T14:15:22Z",
//   "deadlineToAccept": "2019-08-24T14:15:22Z"
// }
//

public class Contract
{
    public string? Id { get; set; }
    public string? FactionSymbol { get; set; }
    public string? Type { get; set; }
    public ContractTerms? Terms { get; set; }
    public bool Accepted { get; set; }
    public bool Fulfilled { get; set; }
    public DateTime Expiration { get; set; }
    public DateTime DeadlineToAccept { get; set; }
}

File: backend/types/contract.go
package types

import "time"

// Different types of contracts that can be created
const (
	CONTRACT_PROCUREMENT = "PROCUREMENT"
	CONTRACT_TRANSPORT   = "TRANSPORT"
	CONTRACT_SHUTTLE     = "SHUTTLE"
)

type ContractPayment struct {
	OnAccepted  int `json:"onAccepted"`
	OnFulfilled int `json:"onFulfilled"`
}

type ContractDeliverGood struct {
	TradeSymbol       string `json:"tradeSymbol"`
	DestinationSymbol string `json:"destinationSymbol"`
	UnitsRequired     int    `json:"unitsRequired"`
	UnitsFufilled     int    `json:"unitsFufilled"`
}

type ContractTerms struct {
	Deadline     time.Time             `json:"deadline"`
	Payment      ContractPayment       `json:"payment"`
	DeliverGoods []ContractDeliverGood `json:"deliver"`
}

type Contract struct {
	ID               string        `json:"id"`
	FactionSymbol    string        `json:"factionSymbol"`
	Type             string        `json:"type"`
	Terms            ContractTerms `json:"terms"`
	Accepted         bool          `json:"accepted"`
	Fulfilled        bool          `json:"fulfilled"`
	Expiration       time.Time     `json:"expiration"`
	DeadlineToAccept time.Time     `json:"deadlineToAccept"`
}
